### ISSUE TITLE
[SRD0039] False positive on named CTE

### ISSUE THREAD

**Issue by sergey-rybalkin** [ErikEJ/SqlServer.Rules#258]
First of all, thank you for the great product, it is a great addition to the SSDT ecosystem.

Unfortunately, looks like there is a problem with `ObjectTwoPartNameRule` analyzer, see the following code sample:

```
WITH cte AS
(
	SELECT c1, c2 FROM dbo.t1
)
SELECT c.c1, c.c2
FROM cte c -- should not trigger an error, but does
```

It should not require schema on the CTE name, however I'm getting the following warning on the last line of code: ` warning SRD0039: SqlServer.Rules : Use fully qualified object names in SELECT, UPDATE, DELETE, MERGE and EXECUTE statements. [schema].[name].`. If I remove `c` as a CTE name alias then everything works as expected. Does this mean that analyzer is not handling named CTE references properly, or am I missing something?
